File: DaySix/JavaScript/task6.js
const prompt = require("readline-sync");

function getAge(userDob) {
    let daySubstring = userDob.substring(0, 2);
    let monthSubstring = userDob.substring(3, 5);
    let yearSubstring = userDob.substring(6);

    let dayInput = parseInt(daySubstring);
    let monthInput = parseInt(monthSubstring);
    let yearInput = parseInt(yearSubstring);

    let currentDate = new Date();
    let currentYear = currentDate.getFullYear();
    let currentMonth = currentDate.getMonth() + 1; // Month starts from 0
    let currentDay = currentDate.getDate();

    let ageYear = currentYear - yearInput;

    if (currentMonth < monthInput || (currentMonth === monthInput && currentDay < dayInput)) {
        ageYear -= 1;
    }

    return ageYear;
}
let user_input = prompt("Enter your date of birth: ");
console.log(getAge(user_input));

File: DaySix/Python/task6.py
from datetime import datetime

def get_age(user_dob):
    day_input = int(user_dob[0:2])
    month_input = int(user_dob[3:5])
    year_input = int(user_dob[6:])

    current_date = datetime.now()
    current_year = current_date.year
    current_month = current_date.month
    current_day = current_date.day

    age_year = current_year - year_input

    if current_month < month_input or (current_month == month_input and current_day < day_input):
        age_year -= 1

    return age_year

user_input = input("Enter your date of birth: ")
print(get_age(user_input))